src/hardware/memory.cpp

```diff
@@ -426,6 +426,15 @@ bool MEM_A20_Enabled(void) {
 	return memory.a20.enabled;
 }
 
+// Use this function for initialization since the public 
+// function is optimized to return on the same state 
+static void InitA20() {
+	const auto enabled = false;
+	const uint32_t phys_base=enabled ? (1024/4) : 0;
+	for (int i=0;i<16;i++) PAGING_MapPage((1024/4)+i,phys_base+i);
+	memory.a20.enabled=enabled;
```

Comment:
Can we bake-out the duplicate function body somehow?

![2022-06-19_17-34](https://user-images.githubusercontent.com/1557255/174506627-1e867236-4e70-43dc-8bfc-642a54dcfeee.png)

Hmm..

``` c++
static void InitA20() {
	memory.a20.enabled = true;
	MEM_A20_Enable(false);
}
```

Reply:
Or give it a full back-and-fourth scrub (if we don't want to meddle with internal state):

``` c++
	MEM_A20_Enable(true); // a bit wasteful ...
	MEM_A20_Enable(false);
```

Reply:
> Can we bake-out the duplicate function body somehow?

See my comment above.

Reply:
>  I felt dirty doing that - it seems like a "code WTF?"

I hear you; but the idea of a dense duplicate code block like that doesn't sit well.

_"DRY might still be better than dirty"_, if I had to choose :-).  

At this point.. we're playing rock paper scissors :smile:

Reply:
let me generate some assembly with the "true -> false" scenario

Reply:
> > I felt dirty doing that - it seems like a "code WTF?"
> 
> I hear you; but the idea of a dense duplicate code block like that doesn't sit well.

If both of you are OK with the "true->false" scenario, I will add a hefty comment explaining it!